webdiff/app.py

```diff
@@ -279,11 +282,19 @@ def run():
 
     PORT = pick_a_port(parsed_args)
 
-    sys.stderr.write('''Serving diffs on http://localhost:%s
+    if app.config.get('USE_HOSTNAME'):
+        _hostname = platform.node()
+        # platform.node will return empty string if it can't find the hostname
+        if not _hostname:
+            sys.stderr.write('Warning: hostname could not be determined')
+        else:
+            HOSTNAME = _hostname
+
+    sys.stderr.write('''Serving diffs on http://%s:%s
 Close the browser tab or hit Ctrl-C when you're done.
-''' % PORT)
+''' % (HOSTNAME, PORT))
     Timer(0.1, open_browser).start()
-    app.run(host='0.0.0.0', port=PORT)
+    app.run(host=HOSTNAME, port=PORT)
```

Comment:
No critical reason, it just makes the code simpler/easier to read.
The ```HOSTNAME``` variable is used for both the app endpoint and to log to the user the url of the webdiff, and I find ```localhost``` a bit nicer and friendlier to display to a user so I went with that instead of 127.0.0.1 or 0.0.0.0 (the latter having a slightly different definition than the former).